app/src/androidTest/java/com/github/se/cyrcle/ui/profile/ProfileScreenMock.kt

```diff
@@ -0,0 +1,303 @@
+import com.github.se.cyrcle.di.mocks.MockImageRepository
+import com.github.se.cyrcle.di.mocks.MockParkingRepository
+import com.github.se.cyrcle.di.mocks.MockUserRepository
+import com.github.se.cyrcle.model.parking.Location
+import com.github.se.cyrcle.model.parking.Parking
+import com.github.se.cyrcle.model.parking.ParkingCapacity
+import com.github.se.cyrcle.model.parking.ParkingProtection
+import com.github.se.cyrcle.model.parking.ParkingRackType
+import com.github.se.cyrcle.model.parking.ParkingViewModel
+import com.github.se.cyrcle.model.parking.TestInstancesParking
+import com.github.se.cyrcle.model.user.User
+import com.github.se.cyrcle.model.user.UserViewModel
+import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.mapbox.geojson.Point
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+
+@RunWith(AndroidJUnit4::class)
+class ProfileScreenMock {
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  private lateinit var mockNavigationActions: NavigationActions
+  private lateinit var mockUserRepository: MockUserRepository
+  private lateinit var mockParkingRepository: MockParkingRepository
+  private lateinit var mockImageRepository: MockImageRepository
+
+  private lateinit var userViewModel: UserViewModel
+  private lateinit var parkingViewModel: ParkingViewModel
+
+  @Before
+  fun setUp() {
+    mockNavigationActions = mock(NavigationActions::class.java)
+    mockUserRepository = MockUserRepository()
+    mockParkingRepository = MockParkingRepository()
+    mockImageRepository = MockImageRepository()
+
+    val user =
+        User(
+            userId = "1",
+            username = "janesmith",
+            firstName = "Jane",
+            lastName = "Smith",
+            email = "jane.smith@example.com",
+            profilePictureUrl = "http://example.com/jane.jpg")
+
+    userViewModel = UserViewModel(mockUserRepository, mockParkingRepository)
+    parkingViewModel = ParkingViewModel(mockImageRepository, mockParkingRepository)
+
+    userViewModel.addUser(user)
+
+    // parking1 already added by whoever created parkingviewmodelmock on instanciation
+    parkingViewModel.addParking(TestInstancesParking.parking2)
+    parkingViewModel.addParking(TestInstancesParking.parking3)
+
+    // Fetch the user
+    userViewModel.getUserById("1")
+
+    // Add favorite parkings
+    userViewModel.addFavoriteParkingToSelectedUser(TestInstancesParking.parking1.uid)
+    userViewModel.addFavoriteParkingToSelectedUser(TestInstancesParking.parking2.uid)
+    userViewModel.addFavoriteParkingToSelectedUser(TestInstancesParking.parking3.uid)
+    // Fetch favorite parkings
+    userViewModel.getSelectedUserFavoriteParking()
+
+    composeTestRule.setContent {
+      ViewProfileScreen(navigationActions = mockNavigationActions, userViewModel = userViewModel)
+    }
+  }
+
+  @Test
+  fun testUserSetupAndFavoriteParkings() {
```

Comment:
Because something was wrong with the user view model, which I then fixed thanks to this test. But you're right, I'll still remove it because now everything's good and it's irrelevant.